internal/platform/state/state.go

```diff
@@ -62,6 +62,11 @@ type OwnershipSyncState interface {
 
 	SetLastOwnershipBlock(block model.Block) error
 	GetLastOwnershipBlock() (model.Block, error)
+	GetOwnershipBlock(blockNumber uint64) (model.Block, error)
+	SetOwnershipBlock(blockNumber uint64, block model.Block) error
+	GetAllStoredBlockNumbers() ([]uint64, error)
+	CleanStoredBlockNumbers() error
```

Comment:
I think that `CleanStoredBlockNumbers` and `DeleteStoredBlockNumbersNewerThanBlockNumber` in essence do the same function. Maybe we should have only one functions which name would be 

`DeleteBlocksForBlockRange(startBlock, endBlock uint64)`